code/modules/nano/modules/camera_monitor.dm

```diff
@@ -0,0 +1,188 @@
+	networks["Research"] = list(access_rd,access_hos,access_captain)
+	networks["Prison"] = list(access_hos,access_captain)
+	networks["Labor"] = list(access_hos,access_captain)
+	networks["Interrogation"] = list(access_hos,access_captain)
+	networks["Atmosphere Alarms"] = list(access_ce,access_hos,access_captain)
+	networks["Fire Alarms"] = list(access_ce,access_hos,access_captain)
+	networks["Power Alarms"] = list(access_ce,access_hos,access_captain)
+	networks["Supermatter"] = list(access_ce,access_hos,access_captain)
+	networks["MiniSat"] = list(access_rd,access_hos,access_captain)
+	networks["Singularity"] = list(access_ce,access_hos,access_captain)
+	networks["Anomaly Isolation"] = list(access_rd,access_hos,access_captain)
+	networks["Toxins"] = list(access_rd,access_hos,access_captain)
+	networks["Telepad"] = list(access_rd,access_hos,access_captain)
+	networks["TestChamber"] = list(access_rd,access_hos,access_captain)
+	
+/datum/nano_module/camera_monitor/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1, var/datum/topic_state/state = default_state)
+	var/data[0]
+
+	data["current"] = null
+
+	if(camera_cache_id != cache_id)
+		cache_id = camera_cache_id
+		cameranet.process_sort()
+
+		var/cameras[0]
+		for(var/obj/machinery/camera/C in cameranet.cameras)
+			if(!can_access_camera(C))
+				continue
+
+			var/cam = C.nano_structure()
+			cameras[++cameras.len] = cam
+
+		camera_cache=list2json(cameras)
+
+	tempnets.Cut()
+	access = list(access_captain) // Assume captain level access when AI
+
+	// Loop through the ID's permission, and check which networks the ID has access to.
+	for(var/l in networks) // Loop through networks.
+		for(var/m in networks[l]) // Loop through access levels of the networks.
+			if(m in access)
+				if(l in network) // Checks if the network is currently active.
+					tempnets.Add(list(list("name" = l, "active" = 1)))
+				else
+					tempnets.Add(list(list("name" = l, "active" = 0)))
+				break
+	if(tempnets.len)
+		data["networks"] = tempnets
+
+	if(current)
+		data["current"] = current.nano_structure()
+	data["cameras"] = list("__json_cache" = camera_cache)
+
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "sec_camera.tmpl", "Camera Console", 900, 800)
+
+		// adding a template with the key "mapContent" enables the map ui functionality
+		ui.add_template("mapContent", "sec_camera_map_content.tmpl")
+		// adding a template with the key "mapHeader" replaces the map header content
+		ui.add_template("mapHeader", "sec_camera_map_header.tmpl")
+
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+		
+/datum/nano_module/camera_monitor/Topic(href, href_list)
+	if(..()) 
+		return 1
+
+	if(href_list["switchTo"])
+		if(usr.stat || ((get_dist(usr, src) > 1 || !( usr.canmove ) || usr.blinded) && !istype(usr, /mob/living/silicon))) return
```

Comment:
Not sure how close Bay and Paradise NanoUI is these days but rather than ```src``` you probably want to use ```nano_host()``` in this case.